unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/WorldState.cs

```diff
@@ -36,82 +32,48 @@ public void Initialize()
             globalSceneId = null;
             currentSceneId = null;
             readyScenes = new HashSet<string>();
-            loadedScenes = new Dictionary<string, ParcelScene>();
-            scenesSortedByDistance = new List<ParcelScene>();
+            loadedScenes = new Dictionary<string, IParcelScene>();
+            scenesSortedByDistance = new List<IParcelScene>();
         }
 
-        public string TryToGetSceneCoordsID(string id)
+        public T GetScene<T>(string id)
```

Comment:
Do we really need this as a Generic? can't we just return an IParcelScene always?

Reply:
We can but we have to add extra casting if we want to use `ParcelScene` explicitly. Just removed the generic to simplify because we aren't using this on many places anyways.